specification/dartLangSpec.tex

```diff
@@ -12287,24 +12291,33 @@ \subsubsection{For-in}
 \LMLabel{for-in}
 
 \LMHash{}%
-Let $D$ be derived from \syntax{<finalConstVarOrType>?}
-and let $n0$ be an identifier that does not occur anywhere in the program.
-A for statement of the form \code{\FOR{} ($D$ \id{} \IN{} $e$) $s$} is equivalent to the following code:
+Let $D$ be derived from \syntax{<finalConstVarOrType>?}.
+A for statement of the form \code{\FOR{} ($D$ \id{} \IN{} $e$)\,\,$S$}
+is then treated as the following code,
+where $\id_1$ and $\id_2$ are fresh identifiers:
 
 \begin{normativeDartCode}
-\VAR{} $n0$ = $e$.iterator;
-\WHILE{} ($n0$.moveNext()) \{
-\ \ $D$ \id{} = $n0$.current;
-\ \ $s$
+$T$ $\id_1$ = $e$;
+\VAR{} $\id_2$ = $id_1$.iterator;
+\WHILE{} ($\id_2$.moveNext()) \{
+\ \ $D$ \id{} = $\id_2$.current;
+\ \ $S$
```

Comment:
Put `S` into a block, `{ S }`. 
Otherwise the `id` declaration is moved to the same level as the $S$, and that would cause a conflict for degenerate examples like `for (var id in []) var id = 2;`

Reply:
Done.